dashboard/app/models/plc/user_course_enrollment.rb

```diff
@@ -28,6 +28,33 @@ class Plc::UserCourseEnrollment < ActiveRecord::Base
 
   after_create :create_enrollment_unit_assignments
 
+  def self.enroll_users(user_emails, course_id)
+    course = Plc::Course.find(course_id)
+    enrolled_users = []
+    nonexistant_users = []
+    nonteacher_users = []
+    other_failure_users = []
+
+    user_emails.each do |email|
+      user = User.find_by_email_or_hashed_email(email)
+
+      if user.nil?
+        nonexistant_users << email
+      elsif !user.teacher?
+        nonteacher_users << email
+      else
+        enrollment = find_or_create_by(user: user, plc_course: course)
+        if enrollment.valid?
+          enrolled_users << email
+        else
+          other_failure_users << email
```

Comment:
In this other case, do we have more details as to why it's invalid? Should we display the `enrollment.errors` to user?